test_allowedflare.py

```diff
@@ -0,0 +1,81 @@
+from jwt import encode
+from cryptography.hazmat.primitives.asymmetric.rsa import generate_private_key
+from cryptography.hazmat.backends import default_backend
+from datetime import UTC, datetime
+
+from allowedflare import authenticate, clean_username
+
+
+def test_clean_username_unmodified(monkeypatch):
+    monkeypatch.delenv('ALLOWEDFLARE_EMAIL_DOMAIN', raising=False)
+    monkeypatch.delenv('ALLOWEDFLARE_PRIVATE_DOMAIN', raising=False)
+    assert clean_username('user@domain.com') == 'user@domain.com'
+
+    monkeypatch.setenv('ALLOWEDFLARE_EMAIL_DOMAIN', 'off')
+    monkeypatch.setenv('ALLOWEDFLARE_PRIVATE_DOMAIN', 'domain.com')
+    assert clean_username('user@domain.com') == 'user@domain.com'
+
+
+def test_clean_username_email_domain_removed(monkeypatch):
+    monkeypatch.setenv('ALLOWEDFLARE_PRIVATE_DOMAIN', 'domain.dev')
+    monkeypatch.setenv('ALLOWEDFLARE_EMAIL_DOMAIN', 'domain.com')
+
+    assert clean_username('user@domain.com') == 'user'
+
```

Comment:
Why is the first `setenv` related to the test? Also, the test name is "email domain removed", which I don't understand where the removal is taking place (I just see `setenv`)

Reply:
Good point will remove.